File: Shared/src/main/java/nju/classroomassistant/shared/log/Logger.java
package nju.classroomassistant.shared.log;

import java.time.LocalDateTime;

public interface Logger {
    default String getLoggerTag() {
        return getClass().getSimpleName();
    }

    default void log(String message, LogSeverity severity) {
        System.out.println(String.format("[%s] [%s] [%s] %s", LocalDateTime.now().toString(), getLoggerTag(), severity, message));
    }

    default void verbose(String message) {
        log(message, LogSeverity.VERBOSE);
    }

    default void warning(String message) {
        log(message, LogSeverity.WARNING);
    }

    default void error(String message) {
        log(message, LogSeverity.ERROR);
    }


}

File: shared/src/main/kotlin/nju/classroomassistant/shared/log/Logger.kt
package nju.classroomassistant.shared.log

enum class LogSeverity {
    VERBOSE, WARNING, ERROR
}

interface Logger {
    val loggerTag: String
        get() = javaClass.simpleName

    fun log(message: String, severity: LogSeverity) {
        println("[$loggerTag] [$severity] $message")
    }

    fun verbose(message: String) {
        log(message, LogSeverity.VERBOSE)
    }

    fun warning(message: String) {
        log(message, LogSeverity.WARNING)
    }

    fun error(message: String) {
        log(message, LogSeverity.ERROR)
    }
}


